bundle.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
+)
+
+func unarchiveBundle(bundleURL, targetDir string) (err error) {
+	url, err := url.Parse(bundleURL)
+	if err != nil {
+		return fmt.Errorf("%s", err)
+	}
+
+	if url.Scheme != "https" {
+		return fmt.Errorf("invalid URL scheme: %s, expecting https", url.Scheme)
+	}
+
+	const storageHost = "storage.googleapis.com"
+	if url.Host != storageHost {
+		return fmt.Errorf("invalid hostname, expecting %s", storageHost)
+	}
+
+	const sep = "/"
+	pathParts := strings.Split(strings.TrimLeft(url.EscapedPath(), sep), sep)
```

Comment:
could you please comment an example url in the code to make it easier understand the code